### PR TITLE
refactor export.py to support customize the data_shape of mindir

### PR DESCRIPTION
**Pull request by HaoyangLee**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [x] You have added unit tests

## Motivation
- User can set the `data_shape` when exporting mindir.
- Remove the functionality of exporting all models in `list_models()`  at once while running export.py, which is not very frequently used when we have many models in `list_models()`.
- Update DBNet++ mindir file, which is exported using data shape [1152, 2048] and achieves the same F-score in lite inference as in eval. The former mindir is exported using data shape [736, 1280], having the lower F-score in lite inference then in eval.

### CODE REVIEW

File: tools/export.py [hunk 1]
```diff
@@ -22,51 +31,31 @@
...
-            raise ValueError("Arg 'model_name' is empty. Please set 'model_name' if 'local_ckpt_path' is not None.")
-        if args.model_name not in list_models():
-            raise ValueError(f"Invalid 'model_name': {args.model_name}. 'model_name' must be one of names in {list_models()}.")
-        if not os.path.isfile(args.local_ckpt_path):
-            raise ValueError(f"No such ckpt file in this path: {args.local_ckpt_path}.")
-    else:  # download online ckpt
-        if args.model_name and args.model_name not in list_models():
-            raise ValueError(f"Invalid 'model_name': {args.model_name}. 'model_name' must be empty or one of names in {list_models()}.")
+    if args.model_name not in list_models():
+        raise ValueError(f"Invalid arg 'model_name': {args.model_name}. 'model_name' must be empty or one of names in {list_models()}.")
```
Comment (hadipash) [inline tools/export.py:54]: adding a flag `choices=list_models()` to `--model_name` will eliminate the need to explicitly check if the model is on the list.

File: tools/export.py [hunk 2]
```diff
@@ -22,51 +31,31 @@
...
+    if args.local_ckpt_path and not os.path.isfile(args.local_ckpt_path):
+        raise ValueError(f"Local ckpt file {args.local_ckpt_path} does not exist. Please check arg 'local_ckpt_path'.")
+    if args.save_dir and not os.path.isdir(args.save_dir):
```
Comment (hadipash) [inline tools/export.py:57]: Can we just create a directory if it doesn't exist?

**Review by hadipash — APPROVED** [review]

File: tools/export.py [hunk 3]
```diff
@@ -22,51 +31,31 @@
...
+        raise ValueError(f"Directory {args.save_dir} does not exist. Please check arg 'save_dir'.")
```
Comment (hadipash) [inline tools/export.py:58]: Can we just create a directory if it doesn't exist?

Comment (HaoyangLee) [reply tools/export.py]: updated

File: tools/export.py (hunk 1 above)
Comment (HaoyangLee) [reply tools/export.py]: updated

**Review by SamitHuang — APPROVED** [review]

File: tests/ut/test_mindir_export.py [hunk 4]
```diff
@@ -6,37 +6,37 @@
 from mindocr import list_models, build_model
 from tools.export import export
 
-@pytest.mark.parametrize('name', ['dbnet_resnet50', 'crnn_resnet34'])
-def test_mindir_infer(name):
+@pytest.mark.parametrize('model_name', list_models())
```
[the remark this answers is not in the document]
Comment (HaoyangLee) [reply tests/ut/test_mindir_export.py]: modified

**Pull request merged**